### ISSUE TITLE
Help understand results: Show estimate for chance of being above random

### ISSUE THREAD

**Issue by torronen** [dotnet/machinelearning-modelbuilder#1874]
**Background**
The "ethics of AI" is increasing in importance, especially in government projects. Tools like Model Builder make building machine learning models very fast and easy which could lead to de-emphasis on testing and validation. Low-performing models have ethical implications in certain cases as it may influence decision making. 

While tracking a potential bug, I am running a random values dataset with only 1000 items (Excel random function for range 0...1). Model Builder reports 60.7% accuracy after a short run which could easily be mistaken as good enough. It probably gets slightly better with time (now 62.22%)

![image](https://user-images.githubusercontent.com/26261427/138869586-ca4002ad-29dd-4525-99d5-e62e4ddf8131.png)

** Suggestion **
Model Builder could report one more metric about the chance of accuracy being above random. It needs someone smarter than me to create the equation, but it probably should be based on number of examples, size of validation data and number of experiments and the range of the results from different experiments. 

Or, it could be just reported as low/medium/high based on some experiments with random datasets.

Or, it could be just a warning about low number of examples if the accuracy is relatively low.

**Additional context**
A challenge is how to deal with overfitted data. Adding this metric could create false trust in the results if the metric shows high change of being above random. 

[random-dataset.csv](https://github.com/dotnet/machinelearning-modelbuilder/files/7417663/random-dataset.csv)

**Comment by LittleLittleCloud** [comment]
@briacht 

Thanks for the suggestions, and something to add

Accuracy is a bad metric to default with especially in binary classification, the least thing we can do is default to AUC, which will be more indicative.

**Comment by torronen** [comment]
This article discusses uncertainty in small datasets https://towardsdatascience.com/small-data-large-uncertainty-1d4059ced4f8 As I recall, validation data set size depends on the size of the full dataset in Model builder and is not a fixed percentage. That makes me think that it might be a good idea to show how many items were used in the validation set. It could help users to estimate the uncertainty even if it interpretation would be left for user to do manually.

**Comment by briacht** [comment]
This is a great point!

@romalpani This will be something to look into for the UI as well -- how do we tell users - "Hey, it looks like you trained on a small dataset; even though your accuracy is high, this may not be a good model"

@LittleLittleCloud I think it's ok to keep Accuracy as default. A lot more people (without data science knowledge) will know what Accuracy means over AUC. But this does bring back the point that we need to add the option for being able to choose another optimizing evaluation metric -- and also someplace for people to be able to see all the different evaluation metrics (maybe we start this in the .mbconfig and then later incorporate into the UI when we can get designs?)

**Comment by torronen** [comment]
One thing here has been bothering me. How is it possible to reach ~63% accuracy with 1000 random values, and especially with Random Forest and not FastTree? Decision tree probably could probably choose any decision points since values should be spread out randomly, and then the one matching validation data would be selected. I thought FastForest should average it out.

That makes me wonder if the way size of the validation data set is selected should be reviewed? Potentially, it might need to be bigger for small datasets. There could be something wrong with randomness in my data, too. 

At least this is something to keep my mind occupied at nights :)
